File: zoom/src/zoomCocoa/ZoomDownloadView.m
//
//  ZoomDownloadView.m
//  ZoomCocoa
//
//  Created by Andrew Hunter on 13/10/2007.
//  Copyright 2007 Andrew Hunter. All rights reserved.
//

#import "ZoomDownloadView.h"


@implementation ZoomDownloadView

- (id)initWithFrame:(NSRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
		// Set up the image
		downloadImage = [[NSImage imageNamed: @"IFDB-downloading"] retain];
		
		// Set up the progress indicator
		progress = [[NSProgressIndicator alloc] initWithFrame: NSMakeRect(NSMinX(frame)+37, NSMinY(frame) + 24, frame.size.width-74, 16)];
		[progress setAutoresizingMask: NSViewWidthSizable|NSViewMaxYMargin];
		[progress setUsesThreadedAnimation: NO];
		
		[self addSubview: progress];
    }
    return self;
}

- (void) dealloc {
	[downloadImage release];
	
	[super dealloc];
}

- (void)drawRect:(NSRect)rect {
	NSSize imageSize = [downloadImage size];
	NSRect bounds = [self bounds];
	
	[[NSColor clearColor] set];
	NSRectFill(bounds);
	
	NSRect downloadRect;
	downloadRect.origin.x = NSMinX(bounds) + (bounds.size.width - imageSize.width) / 2;
	downloadRect.origin.y = NSMinY(bounds) + (bounds.size.height - imageSize.height) / 2;
	downloadRect.size = imageSize;
	[downloadImage drawInRect: downloadRect
					 fromRect: NSMakeRect(0,0, imageSize.width,imageSize.height)
					operation: NSCompositeSourceOver
					 fraction: 1.0];
}

- (BOOL) isOpaque {
	return NO;
}

- (NSProgressIndicator*) progress {
	return progress;
}

@end

File: zoom/src/zoomCocoa/ZoomDownloadView.swift
//
//  ZoomDownloadView.swift
//  Zoom
//
//  Created by C.W. Betts on 9/24/21.
//

import Cocoa

class DownloadView : NSView {
	/// The background image
	private let downloadImage = NSImage(named: "IFDB-downloading")!
	/// The download progress indicator
	@objc let progress: NSProgressIndicator
	
	override init(frame: NSRect) {
		// Set up the progress indicator
		progress = NSProgressIndicator(frame: NSRect(x: frame.minX + 37, y: frame.minY + 24, width: frame.width - 74, height: 16))
		progress.autoresizingMask = [.width, .maxYMargin]
		progress.usesThreadedAnimation = false
		super.init(frame: frame)
		
		addSubview(progress)
	}
	
	required init?(coder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}
	
	override func draw(_ dirtyRect: NSRect) {
		let imageSize = downloadImage.size
		let bounds = self.bounds

		NSColor.clear.set()
		bounds.fill()
		
		var downloadRect = NSRect()
		downloadRect.origin.x = bounds.minX + (bounds.width - imageSize.width) / 2
		downloadRect.origin.y = bounds.minY + (bounds.height - imageSize.height) / 2
		downloadRect.size = imageSize
		downloadImage.draw(in: downloadRect, from: .zero, operation: .sourceOver, fraction: 10)
	}
	
	override var isOpaque: Bool {
		return false
	}
}
